Dockerfile

```diff
@@ -1,17 +1,37 @@
+ARG JMM_STRATEGY=copy
 FROM node:18-alpine AS base
+WORKDIR /app
+
+FROM base as copy
 
-FROM base as downloader
+COPY ./compiler compiler
+
+FROM base as download
 
 ARG JMM_URL
 RUN wget ${JMM_URL} -O compiler.zip
 
 RUN unzip compiler.zip -d compiler
 
+FROM gradle:alpine as git
+WORKDIR /app
+
+RUN apk add --no-cache git
+
+ARG JMM_URL
+ARG JMM_BRANCH
+RUN git clone --single-branch ${JMM_BRANCH:+--branch $JMM_BRANCH} ${JMM_URL} compiler_git
+
+RUN cd compiler_git && gradle installDist
+
+RUN mkdir -p compiler && cp -r compiler_git/build/install/jmm compiler/
```

Comment:
Regarding this, when I bundle my compiler to deploy, I also have to bundle the config.properties file because I use TestUtils. How can we get around this? Is there a way to tell gradle to copy the config.properties? Or is there a way to tell TestUtils which class to use?

Reply:
That's something I hadn't considered, as my compiler has a custom report handling function.
I'm sure we can just copy the config.properties manually as it seems like a common use case.

I'm curious tho, does it not work without it?
The `noErrors` function from `TestUtils` does not use the config at all.

Reply:
You're right, the TestUtils probably doesn't use the config.properties, but my compiler doesn't work without it... I'll try to figure out what is happening and fix it